tests/performance-tests/src/tests/public-api/data-set-query/publicApiDataSetQuery.test.ts

```diff
@@ -0,0 +1,249 @@
+/* eslint-disable no-console */
+import exec from 'k6/execution';
+import { Counter, Rate, Trend } from 'k6/metrics';
+import getEnvironmentAndUsersFromFile from '../../../utils/environmentAndUsers';
+import loggingUtils from '../../../utils/loggingUtils';
+import { isRefinedResponse, pickRandom } from '../../../utils/utils';
+import createPublicApiService, {
+  DataSetMeta,
+  Publication,
+} from '../../../utils/publicApiService';
+import grafanaService from '../../../utils/grafanaService';
+import createQueryGenerator from './queryGenerators';
+import config, { Config } from './config';
+import {
+  logErrorObject,
+  logErrorResponse,
+  logPublicationsAndDataSets,
+  logQueryResponse,
+  logTestStart,
+} from './logging';
+
+const name = 'publicApiDataSetQuery.test.ts';
+
+export const individualQuerySpeedTrend = new Trend(
+  'ees_public_api_individual_query_speed',
+  true,
+);
+export const individualQueryRequestCount = new Counter(
+  'ees_public_api_individual_query_request_count',
+);
+export const individualQueryCompleteCount = new Counter(
+  'ees_public_api_individual_query_complete_count',
+);
+export const queryFailureCount = new Counter(
+  'ees_public_api_query_failure_count',
+);
+export const errorRate = new Rate('ees_errors');
+
+export const timeoutErrorRate = new Rate('ees_dial_io_timeout_count');
+
+export const connectionRefusedErrorRate = new Rate(
+  'ees_public_api_query_connection_refused_count',
+);
+export const connectionResetErrorRate = new Rate(
+  'ees_public_api_query_connection_reset_count',
+);
+export const http2StreamErrorRate = new Rate(
+  'ees_public_api_query_http2_stream_error_count',
+);
+
+export interface PublicationAndDataSets extends Publication {
+  dataSets: (DataSetMeta & {
+    id: string;
+    name: string;
+  })[];
+}
+
+interface SetupData {
+  testConfig: Config;
+  publications: PublicationAndDataSets[];
+}
+
+const environmentAndUsers = getEnvironmentAndUsersFromFile(
+  __ENV.TEST_ENVIRONMENT as string,
```

Comment:
no need for the cast here

Reply:
:+1:  Will look for other instances, but I think I recall that you've cleaned those up already?

Reply:
Yep that's right, there may be one or two but hopefully all the casts for `__ENV` should be gone by now

Reply:
Cool removed this cast, and 2 more that I found, ta.